@here/harp-map-controls/lib/MapControls.ts

```diff
@@ -355,7 +345,10 @@ export class MapControls extends THREE.EventDispatcher {
             this.stopZoom();
         }
 
-        const yawPitchRoll = MapViewUtils.extractYawPitchRoll(this.camera.quaternion);
+        const yawPitchRoll = MapViewUtils.extractYawPitchRoll(
```

Comment:
It is a bit weird, that you need the projection to get yaw, pitch and roll from the camera.
Maybe we have to work on our naming. Yaw, pitch and roll are just the angles of the camera rotation in world space.
If you are referring to the angles relative to the surface or local tangent space we should call them differently, e.g.:
azimuth, tilt and surfaceRoll (unfortunately there is no distinct name for roll).